modules/platforms/cpp/odbc-test/src/utility_test.cpp

```diff
@@ -89,4 +89,58 @@ BOOST_AUTO_TEST_CASE(TestUtilityWriteReadString)
     BOOST_REQUIRE(outStr4.empty());
 }
 
+void CheckDecimalWriteRead(const std::string& val)
+{
+    using namespace ignite::impl::binary;
+    using namespace ignite::impl::interop;
+    using namespace ignite::common;
+    using namespace ignite::utility;
+
+    InteropUnpooledMemory mem(1024);
+    InteropOutputStream outStream(&mem);
+    BinaryWriterImpl writer(&outStream, 0);
+
+    Decimal decimal(val);
+
+    WriteDecimal(writer, decimal);
+
+    outStream.Synchronize();
+
+    InteropInputStream inStream(&mem);
+    BinaryReaderImpl reader(&inStream);
+
+    Decimal out;
+    ReadDecimal(reader, out);
+
+    std::stringstream converter;
+    converter << out;
+
+    std::string res = converter.str();
+
+    BOOST_CHECK_EQUAL(res, val);
+}
+
+/**
+ * Check that Decimal writing and reading works as expected.
+ *
+ * 1. Create Decimal value.
+ * 2. Write using standard serialization algorithm.
+ * 3. Read using standard de-serialization algorithm.
+ * 4. Check that initial and read value are equal.
+ *
+ * Repeat with the following values: 0, 1, -1, 42, -42, 160, -160, 34729864879625196, -34729864879625196
+ */
+BOOST_AUTO_TEST_CASE(TestUtilityWriteReadDecimal)
+{
+    CheckDecimalWriteRead("0");
+    CheckDecimalWriteRead("1");
+    CheckDecimalWriteRead("-1");
+    CheckDecimalWriteRead("42");
+    CheckDecimalWriteRead("-42");
+    CheckDecimalWriteRead("160");
+    CheckDecimalWriteRead("-160");
+    CheckDecimalWriteRead("34729864879625196");
+    CheckDecimalWriteRead("-34729864879625196");
```

Comment:
Should we also test values with fractional parts?

Reply:
Good point. Added.